R/methods-xcmsRaw.R

```diff
@@ -664,6 +672,419 @@ setMethod("findPeaks.centWave", "xcmsRaw", function(object, ppm=25, peakwidth=c(
 })
 
 ############################################################
+## findPeaks.centWaveWithPredictedIsotopeROIs
+setMethod("findPeaks.centWaveWithPredictedIsotopeROIs", "xcmsRaw", function(object, ppm=25, peakwidth=c(20,50), snthresh=10,
+                                                                            prefilter=c(3,100), mzCenterFun="wMean", integrate=1, mzdiff=-0.001,
+                                                                            fitgauss=FALSE, scanrange= numeric(), noise=0, ## noise.local=TRUE,
+                                                                            sleep=0, verbose.columns=FALSE, ROI.list=list(), 
+                                                                            addROIsParams = list("setNewROIs" = FALSE)) {
+  ## perform tradictional peak picking
+  xcmsPeaks <- findPeaks.centWave(
+    object = object, ppm=ppm, peakwidth=peakwidth, snthresh=snthresh,
+    prefilter=prefilter, mzCenterFun=mzCenterFun, integrate=integrate, mzdiff=mzdiff,
+    fitgauss=fitgauss, scanrange=scanrange, noise=noise, ## noise.local=noise.local,
+    sleep=sleep, verbose.columns=TRUE, ROI.list=ROI.list
+  )
+  
```

Comment:
I would suggest that you put all this logic into a function, e.g. `do_add_predicted_isotope_features`. Your `findPeaks.centWaveWithPredictedIsotopeROIs`could then first call `findPeaks.centWave` and then call the `do_add...` on the results.

This would add more flexibility, i.e. the user can first call the `findPeaks.centWave` to identify the peaks and then, in a second call add additional features using `do_add_predicted_isotope_features`. That's also how I try to re-model the `xcms` package, by breaking down the large methods into smaller functions, that are then  easier to maintain, fix and validate. You describe your method also as a two-step approach, thus I think it would be logical to have the logic of the second approach in its own function.